menas/src/main/scala/za/co/absa/enceladus/menas/services/DatasetService.scala

```diff
@@ -316,17 +316,32 @@ class DatasetService @Autowired()(datasetMongoRepository: DatasetMongoRepository
   }
 
   def validateMappingTable(fields: Future[Set[String]],
-                           mt: MappingConformanceRule): RuleValidationsAndFields = {
-    val inputValidation = mt.attributeMappings.values.map(validateInputColumn(fields, _))
+                           mtRule: MappingConformanceRule): RuleValidationsAndFields = {
+    val inputValidation = mtRule.attributeMappings.values.map(validateInputColumn(fields, _))
+    val allOutput = mtRule.allOutputColumns()
+    val outputColumns = mtRule.allOutputColumns().keySet
+
+    val mtFields = for {
+      someMappingTable <- datasetMongoRepository.getConnectedMappingTable(mtRule.mappingTable, mtRule.mappingTableVersion)
+      mtSchema <- someMappingTable.map(mt => datasetMongoRepository.getConnectedSchema(mt.schemaName, mt.schemaVersion)).get
+    } yield mtSchema.map(x => x.fields.flatMap(f => f.getAllChildrenBasePath :+ f.path).toSet ).get
```

Comment:
Most I could do to make it a bit more readable
```suggestion
    val mtFields = for {
      someMappingTable <- datasetMongoRepository.getConnectedMappingTable(mtRule.mappingTable, mtRule.mappingTableVersion)
      mtSchema: Option[Schema] <- someMappingTable.map(mt => datasetMongoRepository.getConnectedSchema(mt.schemaName, mt.schemaVersion)).get
      result = mtSchema.map(_.
        fields.flatMap(f => f.getAllChildrenBasePath :+ f.path).toSet
      ).getOrElse(Set.empty)
    } yield result
```